ui/app/registration/controllers/patientCommonController.js

```diff
@@ -15,6 +15,63 @@ angular.module('bahmni.registration')
             $scope.genderCodes = Object.keys($rootScope.genderMap);
             $scope.dobMandatory = appService.getAppDescriptor().getConfigValue("dobMandatory") || false;
             $scope.readOnlyExtraIdentifiers = appService.getAppDescriptor().getConfigValue("readOnlyExtraIdentifiers");
+            $scope.showSaveConfirmDialogConfig = appService.getAppDescriptor().getConfigValue("showSaveConfirmDialog");
+
+            // Flag variable to check if Dont Save button is pressed or not
+            var flag = 0;
+
+            // Flag to check whether the button has an href or not
+            var hrefFlag = 0;
+
+            // Function is called when the Home Button is Clicked
+            $rootScope.homeNavigate = function () {
+                if ($scope.showSaveConfirmDialogConfig) {
+                    event.preventDefault();
+                    $scope.targetUrl = event.currentTarget.getAttribute('href');
+                    naviConfirmBox(event);
+                }
+            };
+
+            // Checks for State Change
+            var stateChangeListener = $rootScope.$on("$stateChangeStart", function (event, toState, toParams) {
+                naviConfirmBox(event, toState, toParams);
```

Comment:
I think this function could go as for now it is not yet reused, and seems to be quite specific, so not very reusable.

I mean by that that its code should come directly in this scope here..